dips/dip-158.md

```diff
@@ -0,0 +1,794 @@
+
+For completeness purposes, the PaymentActionObject now becomes:
+
+|Field|Type|Required?|Description|
+|-----|----|---------|-----------|
+| amount | uint | Y | Amount of the transfer.  Base units are the same as for on-chain transactions for this currency.  For example, if DiemUSD is represented on-chain where "1" equals 1e-6 dollars, then "1" equals the same amount here.  For any currency, the on-chain mapping must be used for amounts. |
+| currency | enum | Y | One of the supported on-chain currency types, e.g. XUS|
+| action | enum | Y | Populated in the request.  This value indicates the requested action to perform. For immediate capture and P2P transfer, "charge" is used.  For auth and capture, "auth" and "capture" are now available. "capture" can only be performed after a valid "auth" |
+| valid_until | uint | N | Unix timestamp indicating until when the payment can be authorized. Once this time has been reached, the payment cannot be authorized and funds cannot be locked. |
+| timestamp | uint | Y | Unix timestamp indicating the time that the payment Command was created.
+|
+
+```
+{
+    "amount": 100000000,
+    "currency": "XUS",
+    "action": "auth",
+    "valid_until": 74000,
+    "timestamp": 72322,
+}
+```
+
+## Additional Command Types
+The following is a list of new values for the `command_type` field of the `CommandRequestObject`. Each command type denotes a step in the off-chain negotiations and specifies the data of the requests and the response 
+* `InfoCommand` - Allows obtianing payment information based on reference id
+* `InitCommand` - Initializes the payment process
+* `ReadyCommand` - Notifies the payment is ready for settlement on-chain
+* `AuthorizedCommand` - Notifies that payment funds are authorized (locked) for future capture
+* `CaptureCommand` - Request to capture previously authorized (locked) funds 
+* `VoidCommand` - Voids a payment
+* `AbortCommand` - Aborts a payment
+
+Accordingly, the following list specifies additional values for `_ObjectType` field of  `CommandObject`
+* `InfoCommand` 
+* `InitCommand` 
+* `ReadyCommand`
+* `AuthorizedCommand` 
+* `CaptureCommand`
+* `VoidCommand`
+* `AbortCommand`
+
+## PaymentInfoObject (new)
+The minimal payment details required in order for the payer to decide whether the accept the payment or reject it. 
+
+Field|Type|Required?|Description|
+|-|-|-|-|
+|receiver|PaymentActorObject|Y|Only the address field of the PaymentActorObject is required. i.e. status element is not required|
+|action|PaymentActionObject|Y|Information regarding the type of payment and payment amount|
+|reference_id|str|Y|Unique reference ID of this payment. This value should be globaly unique|
+|description|str|N|Description of the payment. To be displayed to the user|
+|
+
+```
+"payment_info": 
+{
+  "receiver": {
+    "address": "dm1pllhdmn9m42vcsamx24zrxgs3qqqpzg3ng32kvacapx5fl",         
+    "merchant_data": { 
+      "name": "Uber",
+      "legal_name": "Uber Technologies Inc.",
```

Comment:
Likewise, it might be better to use fake merchant information

Reply:
Will change.